Math/NumberTheory/ArithmeticFunctions/Inverse.hs

```diff
@@ -108,7 +114,7 @@ invSigma fs
     divs = runFunctionOnFactors divisorsListA fs
 
     n :: a
-    n = product $ map (\(p, k) -> unPrime p ^ k) fs
+    n = product $ map (\(p, k') -> unPrime p ^ k') fs
```

Comment:
While we are here, could you please replace this with `factorBack`?

Reply:
Of course!